### PR TITLE
Gas0hegahoaewishpo

### PR DESCRIPTION
**Pull request by p-fuchs**

### CODE REVIEW

File: main-command/src/main/scala/sbt/internal/server/Server.scala [hunk 1]
```diff
@@ -97,7 +97,8 @@ private[sbt] object Server {
                 case null =>
                 case s    => s.close()
               }
-              log.info(s"sbt server started at ${connection.shortName}")
+              log.info(s"sbt server - TM0 started at ${connection.shortName}")
```
Comment (p-fuchs) [inline main-command/src/main/scala/sbt/internal/server/Server.scala]: TODO: change log

File: main-command/src/main/scala/sbt/internal/server/Server.scala [hunk 2]
```diff
@@ -159,7 +160,8 @@ private[sbt] object Server {
       }
 
       override def shutdown(): Unit = {
-        log.info("shutting down sbt server")
+        log.info("HELLLLLLLO")
```
Comment (p-fuchs) [inline main-command/src/main/scala/sbt/internal/server/Server.scala]: TODO: revert this

File: main-command/src/main/scala/sbt/internal/ui/UITask.scala [hunk 3]
```diff
@@ -11,15 +11,15 @@ package sbt.internal.ui
...
+import sbt.BasicCommandStrings.{ Cancel, Shutdown, TerminateAction }
+import sbt.BasicKeys.{ colorShellPrompt, historyPath }
 import sbt.State
 import sbt.internal.CommandChannel
 import sbt.internal.util.ConsoleAppender.{ ClearPromptLine, ClearScreenAfterCursor, DeleteLine }
 import sbt.internal.util._
-import sbt.internal.util.complete.{ Parser }
+import sbt.internal.util.complete.Parser
 
+//import java.nio.file.{ Files, Paths, StandardOpenOption }
```
Comment (p-fuchs) [inline main-command/src/main/scala/sbt/internal/ui/UITask.scala]: Can we remove?

File: main/src/main/scala/sbt/internal/CommandExchange.scala [hunk 4]
```diff
@@ -482,6 +482,7 @@ private[sbt] final class CommandExchange {
                   case c: NetworkChannel if !c.isInteractive => exit(mt)
                   case _                                     =>
                 }
+
```
Comment (p-fuchs) [inline main/src/main/scala/sbt/internal/CommandExchange.scala:485]: Revrt this line

File: main/src/main/scala/sbt/internal/LogManager.scala [hunk 5]
```diff
@@ -318,7 +319,7 @@ object LogManager {
   // This is the default implementation for the relay appender
   val defaultRelay: Unit => ConsoleAppender = _ => defaultRelayImpl
 
-  private[this] lazy val defaultRelayImpl: ConsoleAppender = new RelayAppender("Relay0")
+  private[this] lazy val defaultRelayImpl = new RelayAppender("Relay0")
```
Comment (p-fuchs) [inline main/src/main/scala/sbt/internal/LogManager.scala:322]: Probably we can revert

File: main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala [hunk 6]
```diff
@@ -422,12 +424,14 @@ object BuildServerProtocol {
             val param = Converter.fromJson[CompileParams](json(r)).get
             val targets = param.targets.map(_.uri).mkString(" ")
             val command = Keys.bspBuildTargetCompile.key
-            val _ = callback.appendExec(s"$command $targets", Some(r.id))
+            callback.log.debug(s"******** param: $param, command: $command")
```
Comment (p-fuchs) [inline main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala]: We can remove

File: main/src/main/scala/sbt/internal/server/NetworkChannel.scala [hunk 7]
```diff
@@ -66,15 +61,15 @@ final class NetworkChannel(
       auth: Set[ServerAuthentication],
       instance: ServerInstance,
       handlers: Seq[ServerHandler],
-      log: Logger
+      log: Logger,
```
Comment (p-fuchs) [inline main/src/main/scala/sbt/internal/server/NetworkChannel.scala]: Maybe remove?

File: main/src/main/scala/sbt/internal/server/NetworkChannel.scala [hunk 8]
```diff
@@ -120,13 +115,21 @@ final class NetworkChannel(
     def jsonRpcRespondError(execId: Option[String], code: Long, message: String): Unit =
       self.respondError(code, message, execId)
 
-    def jsonRpcNotify[A: JsonFormat](method: String, params: A): Unit =
+    def jsonRpcNotify[A: JsonFormat](method: String, params: A): Unit = {
```
Comment (p-fuchs) [inline main/src/main/scala/sbt/internal/server/NetworkChannel.scala]: Revert the `{}`?

File: main/src/main/scala/sbt/internal/server/NetworkChannel.scala [hunk 9]
```diff
@@ -120,13 +115,21 @@ final class NetworkChannel(
...
       self.jsonRpcNotify(method, params)
+    }
 
-    def appendExec(commandLine: String, execId: Option[String]): Boolean =
-      self.append(Exec(commandLine, execId, Some(CommandSource(name))))
+    def appendExec(
+        commandLine: String,
+        execId: Option[String],
+        originId: Option[String] = None
+    ): Boolean = {
```
Comment (p-fuchs) [inline main/src/main/scala/sbt/internal/server/NetworkChannel.scala]: Revert the {}?

File: main/src/main/scala/sbt/internal/server/NetworkChannel.scala [hunk 10]
```diff
@@ -120,13 +115,21 @@ final class NetworkChannel(
...
+      self.append(Exec(commandLine, execId, Some(CommandSource(name)), originId))
+    }
 
-    def appendExec(exec: Exec): Boolean = self.append(exec)
+    def appendExec(exec: Exec): Boolean = {
```
Comment (p-fuchs) [inline main/src/main/scala/sbt/internal/server/NetworkChannel.scala]: Revert the {}?

File: main/src/main/scala/sbt/internal/server/NetworkChannel.scala [hunk 11]
```diff
@@ -176,7 +179,11 @@ final class NetworkChannel(
           try {
             val onHeader: String => Unit = line => {
               if (line.startsWith(ct) && line.contains(x1)) {
-                logMessage("error", s"server protocol $x1 is no longer supported")
+                logMessage(
+                  "error",
+                  s"server protocol $x1 is no longer supported",
+                  Some("run:error")
```
Comment (p-fuchs) [inline main/src/main/scala/sbt/internal/server/NetworkChannel.scala]: Can we remove new line?

File: main/src/main/scala/sbt/internal/server/NetworkChannel.scala [hunk 12]
```diff
@@ -229,15 +236,19 @@ final class NetworkChannel(
             onNotification(ntf)
           } catch {
             case LangServerError(code, message) =>
-              logMessage("error", s"error $code while handling notification: $message")
+              logMessage(
+                "error",
+                s"error $code while handling notification: $message",
+                Some("handle-body")
```
Comment (p-fuchs) [inline main/src/main/scala/sbt/internal/server/NetworkChannel.scala]: Same here

File: protocol/src/main/contraband/bsp.contra [hunk 13]
```diff
@@ -572,6 +582,39 @@ type TestResult {
...
+    displayName: String!
+    message: String
+    status: Int!
+    location: sbt.internal.bsp.Location
+    dataKind: String
+    data: sjsonnew.shaded.scalajson.ast.unsafe.JValue
+}
+
+
+type TestReport {
```
Comment (p-fuchs) [inline protocol/src/main/contraband/bsp.contra:597]: Do we need descriptions for these?

Comment (LaMavia) [reply protocol/src/main/contraband/bsp.contra:597]: I don't think so

File: main/src/main/scala/sbt/internal/server/NetworkChannel.scala (hunk 11 above)
Comment (LaMavia) [reply main/src/main/scala/sbt/internal/server/NetworkChannel.scala]: Can't remove the newline, since it's formatter-dictated

**Pull request closed without merging**